src/ui/components/simple/Frame/index.mjs

```diff
@@ -0,0 +1,53 @@
+import ScriptRegion from '../../abstract/ScriptRegion';
+
+class Frame extends ScriptRegion {
+  constructor(ui, parent) {
+    super(ui);
+
+    this.loading = false;
+    this.shown = false;
+
+    this.parent = parent;
+
+    // TODO: Draw layers
+
+    this.show();
+  }
+
+  preLoadXML(node) {
+    const { id, name } = node.attributes;
+
+    if (name) {
+      this.name = name;
+    }
+
+    if (id) {
+      // TODO: Assign identifier for this frame
+    }
+
+    this.loading = true;
+    this.shown = false;
```

Comment:
Don't forget about `this.visible`, too (when you get around to implementing the rest of `show()`.

Reply:
👍